components/tutorial/tutorial_view.jsx

```diff
@@ -5,59 +5,48 @@ import $ from 'jquery';
 import PropTypes from 'prop-types';
 import React from 'react';
 
-import ChannelStore from 'stores/channel_store.jsx';
-import Constants from 'utils/constants.jsx';
-
 import TutorialIntroScreens from './tutorial_intro_screens.jsx';
 
 export default class TutorialView extends React.Component {
-    constructor(props) {
-        super(props);
-
-        this.handleChannelChange = this.handleChannelChange.bind(this);
-
-        this.state = {
-            townSquare: ChannelStore.getByName(Constants.DEFAULT_CHANNEL)
-        };
-    }
     componentDidMount() {
-        ChannelStore.addChangeListener(this.handleChannelChange);
-
         if (this.props.isRoot) {
             $('body').addClass('app__body');
         }
     }
-    componentWillUnmount() {
-        ChannelStore.removeChangeListener(this.handleChannelChange);
 
+    componentWillUnmount() {
         if (this.props.isRoot) {
             $('body').removeClass('app__body');
         }
     }
-    handleChannelChange() {
-        this.setState({
-            townSquare: ChannelStore.getByName(Constants.DEFAULT_CHANNEL)
-        });
-    }
+
     render() {
         return (
             <div
                 id='app-content'
                 className='app__content'
             >
                 <TutorialIntroScreens
-                    townSquare={this.state.townSquare}
-                    handleDone={this.handleDone}
+                    townSquareDisplayName={this.props.townSquareDisplayName}
+                    appDownloadLink={this.props.appDownloadLink}
+                    isLicensed={this.props.isLicensed}
+                    restrictTeamInvite={this.props.restrictTeamInvite}
+                    supportEmail={this.props.supportEmail}
                 />
             </div>
         );
     }
 }
 
-TutorialView.defaultProps = {
-    isRoot: true
+TutorialView.propTypes = {
```

Comment:
can we make this **static** inside the class?

Reply:
No objection, but would you be amenable to deferring this? I've touched dozens of components, and I'm somewhat reticent to make this particular change across all of them, at least as part of this set of PRs.

Reply:
sure, non blocking